From: Stig Sandbeck Mathisen <ssm@debian.org>
Date: Mon, 30 Jun 2014 16:49:41 +0200
Subject: Do not require rubygems

---
 bin/facter | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/bin/facter b/bin/facter
index 7aec067..c8db6d3 100755
--- a/bin/facter
+++ b/bin/facter
@@ -4,17 +4,6 @@
 # This is primarily for 1.8.7 since 1.9.2+ doesn't put '.' on the load path
 $LOAD_PATH.delete '.'
 
-# Bundler and rubygems maintain a set of directories from which to
-# load gems. If Bundler is loaded, let it determine what can be
-# loaded. If it's not loaded, then use rubygems. But do this before
-# loading any facter code, so that our gem loading system is sane.
-if not defined? ::Bundler
-  begin
-    require 'rubygems'
-  rescue LoadError
-  end
-end
-
 require 'facter/application'
 
 Facter::Application.run(ARGV)
